pkg/process/metadata/workloadmeta/grpc_test.go

```diff
@@ -0,0 +1,271 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package workloadmeta
+
+import (
+	"context"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+)
+
+func assertSetEvent(t *testing.T, expected, actual *pbgo.ProcessEventSet) {
+	t.Helper()
+
+	assert.Equal(t, expected.Pid, actual.Pid)
+	assert.Equal(t, expected.Nspid, actual.Nspid)
+	assert.Equal(t, expected.ContainerId, actual.ContainerId)
+	assert.Equal(t, expected.CreationTime, actual.CreationTime)
+	if expected.Language != nil {
+		assert.Equal(t, expected.Language.Name, actual.Language.Name)
+	}
+}
+
+func assertUnsetEvent(t *testing.T, expected, actual *pbgo.ProcessEventUnset) {
+	t.Helper()
+
+	assert.Equal(t, expected.Pid, actual.Pid)
+}
+
+func toEventSet(proc *procutil.Process) *pbgo.ProcessEventSet {
+	return &pbgo.ProcessEventSet{Pid: proc.Pid}
+}
+
+func toEventUnset(proc *procutil.Process) *pbgo.ProcessEventUnset {
+	return &pbgo.ProcessEventUnset{Pid: proc.Pid}
+}
+
+func TestGetGRPCStreamPort(t *testing.T) {
+	t.Run("invalid port", func(t *testing.T) {
+		cfg := config.Mock(t)
+		cfg.Set("process_config.language_detection.grpc_port", "lorem ipsum")
+
+		assert.Equal(t, config.DefaultProcessEntityStreamPort, getGRPCStreamPort(cfg))
+	})
+
+	t.Run("valid port", func(t *testing.T) {
+		cfg := config.Mock(t)
+		cfg.Set("process_config.language_detection.grpc_port", "1234")
+
+		assert.Equal(t, 1234, getGRPCStreamPort(cfg))
+	})
+
+	t.Run("default", func(t *testing.T) {
+		cfg := config.Mock(t)
+		assert.Equal(t, config.DefaultProcessEntityStreamPort, getGRPCStreamPort(cfg))
+	})
+}
+
+func TestStartStop(t *testing.T) {
+	cfg := config.Mock(t)
+
+	extractor := NewWorkloadMetaExtractor(cfg)
+	cfg.Set("process_config.language_detection.grpc_port", "0") // Tell the os to choose a port for us to reduce flakiness
+	srv := NewGRPCServer(config.Mock(t), extractor)
+
+	err := srv.Start()
+	assert.NoError(t, err)
+
+	assertEventuallyReturns(t, func() {
+		srv.Stop()
+	}, time.Second*5)
```

Comment:
why not just use `assert.Eventually()` instead of writing your own?

Reply:
That's a good idea! I've removed `assertEventuallyReturns` and replaced it with `assert.Eventually()` in https://github.com/DataDog/datadog-agent/pull/17594/commits/c7463db5298ad98d60286fa9c97d43ba876488a3